motioneye/config.py

```diff
@@ -178,10 +179,86 @@ def webcontrol_interface(v, data):
     'netcam_use_tcp': 'rtsp_uses_tcp',
     'text_scale': text_scale,
     'webcontrol_interface': webcontrol_interface,
+    # motion pre-v4.1
     'webcontrol_parms': None,
 }
 
 
+def netcam_keepalive_params(v, data):
+    if v == True:
+        v = 'on'
+
+    elif v == False:
+        v = 'off'
+
+    if 'netcam_params' in data and data['netcam_params']:
+        return {'netcam_params': data['netcam_params'] + ',keepalive = ' + v}
+
+    return {'netcam_params': 'keepalive = ' + v}
+
+
+def netcam_tolerant_check_params(v, data):
```

Comment:
From 4.3 to 4.4, the individual netcam parameter settings need to be merged into `netcam_params` as multiple posts of its value, hence data needs to be mutated (and only for this reason still passed to the functions) when reading in the individual 4.3 settings.

The other way round the parts of the `netcam_ param` value needs to be split again into the individual settings.

This more complex mapping compared to mapping a single key to another single key was while I asked for help here in the first case 🙂.

Reply:
Let me rephrase in an attempt to get my point through, even though it's such a small thing in the end :) 

Why have `netcam_tolerant_check_params` and `netcam_use_tcp_params` been written in different fashion than `netcam_keepalive_params`? All of those three are used for 4.3 -> 4.4 migration, all of those config parameters are to be migrated similarly. Mapping func parameter `v` is used to decide between two values for the v4.4 param, and `data['netcam_params']` (if it exists already) is passed along), yet the function implementation is not the same.
Mostly it is IMO code readability and repetition issue in `netcam_tolerant_check_params` and `netcam_use_tcp_params`. Mutating `v` might not be an real issue at all if it is not used again later by the mapping engine.

To make it clear what I meant, I suggested a change for `netcam_tolerant_check_params` so that it IMO would be more readable (and similar to `netcam_keepalive_params`).

Reply:
Ah sorry for misunderstanding. `netcam_keepalive_params` was handled differently since it can be `force` as well. But you are right, using the same structure for the others (just `else` instead of `elif`) makes it look cleaner. It can be even a one-liner there.

Reply:
Done, and all converted to one-liner, so looks much cleaner now.